mlrun/datastore/filestore.py

```diff
@@ -17,18 +17,34 @@
 
 import fsspec
 
+import mlrun
+
 from .base import DataStore, FileStats
 
 
 class FileStore(DataStore):
     def __init__(self, parent, schema, name, endpoint="", secrets: dict = None):
         super().__init__(parent, name, "file", endpoint, secrets=secrets)
 
+        self._item_path, self._real_path = None, None
+        if mlrun.mlconf.storage.item_to_real_path:
+            # map item path prefix with real (local) path prefix e.g.:
+            # item_to_real_path="\data::c:\\mlrun_data" replace \data\x.csv with c:\\mlrun_data\x.csv
+            split = mlrun.mlconf.storage.item_to_real_path.split("::")
+            self._item_path = split[0].strip()
+            self._real_path = split[1].strip().rstrip("/").rstrip("\\")
+
     @property
     def url(self):
         return self.subpath
 
-    def _join(self, key):
+    def _join(self, key: str):
+        if self._item_path and not self.subpath:
+            if key.startswith(self._item_path):
+                suffix = key[len(self._item_path) :]
+                if suffix[0] in ["/", "\\"]:
+                    suffix = suffix[1:]
+                key = path.join(self._real_path, suffix)
```

Comment:
either comment "e.g.: x/ -> x" or move to a function that its signature reflect the logic